packages/core/strapi/tests/api/relations.test.api.js

```diff
@@ -831,40 +833,58 @@ describe('Relations', () => {
       });
       expect(updatedShop.data).toMatchObject(expectedShop);
     });
-  });
 
-  test('Update relations using the same id multiple times', async () => {
-    const shop = await createShop({
-      anyToManyRel: [
-        { id: id1, position: { end: true } },
-        { id: id2, position: { end: true } },
-      ],
-    });
+    test('Update relations using the same id multiple times', async () => {
+      const shop = await createShop({
+        anyToManyRel: [
+          { id: id1, position: { end: true } },
+          { id: id2, position: { end: true } },
+        ],
+      });
 
-    const updatedShop = await updateShop(shop.data, {
-      anyToManyRel: [
-        { id: id1, position: { end: true } },
-        { id: id1, position: { start: true } },
-        { id: id1, position: { after: id2 } },
-      ],
-    });
+      const updatedShop = await updateShop(shop.data, {
+        anyToManyRel: [
+          { id: id1, position: { end: true } },
+          { id: id1, position: { start: true } },
+          { id: id1, position: { after: id2 } },
+        ],
+      });
 
-    const expectedShop = shopFactory({
-      anyToManyRel: [{ id: id2 }, { id: id1 }],
+      const expectedShop = shopFactory({
+        anyToManyRel: [{ id: id2 }, { id: id1 }],
+      });
+      expect(updatedShop.data).toMatchObject(expectedShop);
     });
-    expect(updatedShop.data).toMatchObject(expectedShop);
-  });
 
-  test('Update relations with invalid connect array in strict mode', async () => {
-    const shop = await createShop({
-      anyToManyRel: [{ id: id1, position: { end: true } }],
-    });
+    test('Update relations with invalid connect array in strict mode', async () => {
+      const shop = await createShop({
+        anyToManyRel: [],
+      });
 
-    // Connect relation before id2, but id2 is not in the DB or in the connect array
-    const updatedShop = await updateShop(shop.data, {
-      anyToManyRel: [{ id: id1, position: { after: id2 } }],
+      // Connect before an id that does not exist.
+      const updatedShop = await updateShop(shop.data, {
+        anyToManyRel: [{ id: id1, position: { after: id2 } }],
+      });
+
+      expect(updatedShop.error).toMatchObject({ status: 400, name: 'ValidationError' });
     });
 
-    expect(updatedShop.error).toMatchObject({ status: 400, name: 'ValidationError' });
+    test('Update relations with invalid connect array in non-strict mode', async () => {
+      const shop = await createShop({
+        anyToManyRel: [{ id: id1 }],
+      });
+
+      // Connect before an id that does not exist.
+      const updatedShop = await updateShop(shop.data, {
+        anyToManyRel: [{ id: id2, position: { after: id3 } }],
+        strictConnect: false,
+      });
```

Comment:
does that work ? Isn't `strictConnect` supposed to be next to `connect/set` ?

Reply:
yes, it's the updateShop method that adds this strict connect attribute to inside all of the conects

Reply:
Ah yes I see, I mixed it with `updateEntry`. Thanks!